### DISCUSSION TITLE
Is it possible to recreate `wireframe` texture in Mitsuba 3?

### DISCUSSION THREAD

**Discussion by jaroslavknotek** [mitsuba-renderer/mitsuba3#1177] [Q&A] [answered]
The [documentation from version 0.4.3 - section 8.6.3](https://www.mitsuba-renderer.org/releases/0.4.3/documentation.pdf) shows a `wireframe` texture. 

I was wondering if it is possible to render a model and reveal its mesh in the same manner in the current version of mitsuba3.

I tried applying `mesh_attribute` texture with random `face_color` the same way it's in the [current documentation](https://mitsuba.readthedocs.io/en/latest/src/generated/plugins_textures.html#id4):
![face_color bunny](https://mitsuba.readthedocs.io/en/latest/_images/texture_mesh_attribute_face.jpg)

It reveals the structure but I don't think it's the same as a wireframe.

**Comment by njroussel** [comment, marked as answer]
Hi @jaroslavknotek 

This hasn't been ported to Mitsuba 3, and there currently isn't any plan to do it either.

That said, I do think that this might still be possible in a custom `mi.Texture` plugin. I don't know how the original implementation worked, but any call to `Texture.eval()` passes a `SurfaceInteraction3f` which should hold enough information to determine whethere or not the current point is close enough to an edge to be rendered with a different color (wireframe color).

**Reply by jaroslavknotek** [reply]
Thank you, I will try that.
